common/cache/lru.go

```diff
@@ -300,20 +313,22 @@ func (c *lru) putInternal(key interface{}, value interface{}, allowUpdate bool)
 		entry.createTime = c.clock.Now().UTC()
 	}
 
-	if len(c.byKey) >= c.maxSize {
+	for c.currSize >= c.maxSize {
 		c.evictOnceInternal()
 	}
-	if len(c.byKey) >= c.maxSize {
+	if c.currSize >= c.maxSize {
 		return nil, ErrCacheFull
 	}
 
 	element := c.byAccess.PushFront(entry)
 	c.byKey[key] = element
+	c.currSize += getSize(value)
```

Comment:
c.currSize can exceed c.maxSize?

Reply:
It will, but items will be evict after next put until currSize is under maxSize.

Reply:
This essentially change the maxSize from a hard limit to a soft limit. Should be ok but I don't see a reason for breaking that if it's easy to keep it it as a hard limit. On L320 the condition can be `c.currSize + size > c.maxSize`, then we get a hard limit?